lib/Subscribers/SendNewSubscriberNotification.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace MailPoet\Subscribers;
+
+use MailPoet\Config\Renderer;
+use MailPoet\Models\Segment;
+use MailPoet\Models\Setting;
+use MailPoet\Models\Subscriber;
+
+class SendNewSubscriberNotification {
+
+  const SENDER_EMAIL_PREFIX = 'wordpress@';
+  const SETTINGS_KEY = 'subscriber_email_notification';
+
+  /** @var Renderer */
+  private $renderer;
+
+  /** @var \MailPoet\Mailer\Mailer */
+  private $mailer;
+
+  /**
+   * @param \MailPoet\Mailer\Mailer|null $mailer
+   * @param Renderer|null $renderer
+   */
+  function __construct($mailer = null, $renderer = null) {
+    if($renderer) {
+      $this->renderer = $renderer;
+    } else {
+      $caching = ! WP_DEBUG;
+      $debugging = WP_DEBUG;
+      $this->renderer = new Renderer($caching, $debugging);
+    }
+    if($mailer) {
+      $this->mailer = $mailer;
+    } else {
+      $this->mailer = new \MailPoet\Mailer\Mailer(false, $this->constructSenderEmail());
+    }
+  }
+
+  /**
+   * @param Subscriber $subscriber
+   * @param Segment[] $segments
+   *
+   * @throws \Exception
+   */
+  function send(Subscriber $subscriber, array $segments) {
+    $settings = Setting::getValue(SendNewSubscriberNotification::SETTINGS_KEY);
+    if($this->isDisabled($settings)) {
+      return;
+    }
+    try {
+      $this->mailer->getSenderNameAndAddress($this->constructSenderEmail());
+      $this->mailer->send($this->constructNewsletter($subscriber, $segments), $settings['address']);
+    } catch(\Exception $e) {
+      if(WP_DEBUG) {
+        throw $e;
+      }
+    }
+  }
+
+  public static function isDisabled($settings) {
+    if(!is_array($settings)) {
+      return true;
+    }
+    if(!isset($settings['enabled'])) {
+      return true;
+    }
+    if(!isset($settings['address'])) {
```

Comment:
This means that we don't send an email if there is no address for subscription notification.
But in admin section it is possible to save form with **yes** but with empty email address.
https://www.dropbox.com/s/syxhlgnzpexp9oe/Screenshot%202018-10-15%2011.59.33.png?dl=0

I think this state is confusing since someone might expect that it would send the notification with some default sender. I think we should either don't allow to save the settings with yes and empty email or implement sending to default sender (+ describe this behavior on settings page).

What do you think?